lib/views/profile.js

```diff
@@ -1,36 +1,46 @@
+const pug = require('pug')
 const config = require('../config')
 const db = require('../db')
-const pug = require('pug')
-const sha1 = require('sha1')
+const { dotget } = require('../util')
+
+const { User } = db.model
 
 module.exports = function (req, res) {
-  if (req.method == 'POST') {
+  if (req.method === 'POST') {
     update(req, res)
   } else {
     display(req, res)
   }
 }
 
-function update (req, res) {
-  db.model.User.findById(req.user.id).then(user => {
-    if (req.body.name) { user.name = req.body.name }
-
-    if (req.body.affiliation) { user.affiliation = req.body.affiliation }
+function hasPasswordChanged ({ password1, password2 }) {
+  return !!(password1 && password2 && password1 === password2)
```

Comment:
This is difficult to read, is there a reason to make it so terse?

Reply:
Yes - I want to be sure that `hasPasswordChanged` returns boolean.

JS has this ugly behavior that when one chain expressions with `&&`, it will return either the most-right expression (considering that the preceding expressions are truthy) or a falsey value (which can be everything - empty string, `null`, `undefined` etc.). Double negation is used to make sure that the function will return a boolean.

Reply:
I think it is okay to keep, but could you add a comment.

Reply:
@3ach @cjmyers I've made the code more explicit in 3eeeaf4